### PR TITLE
Formplayer / Preview: Menus

### PR DESCRIPTION
**Pull request by biyeun**
@benrudolph @wpride cc @orangejenny 

menu-formplayer
menu-preview
menu-responsive

### CODE REVIEW

File: corehq/apps/app_manager/templates/app_manager/partials/preview_app.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
   <iframe
       class="preview-phone-window"
       frameBorder="0"
+      scrolling="no"
```
Comment (benrudolph) [inline corehq/apps/app_manager/templates/app_manager/partials/preview_app.html]: 💯

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/menus/menu_view.js [hunk 2]
```diff
@@ -18,8 +20,37 @@ FormplayerFrontend.module("SessionNavigate.MenuList", function (MenuList, Formpl
...
+            var $pauseBtn = $playBtn.parent().find('.js-module-audio-pause');
+            $pauseBtn.removeClass('hide');
+            $playBtn.addClass('hide');
+            var $audioElem = $playBtn.parent().find('.js-module-audio');
+            if ($audioElem.data('isFirstPlay') !== 'yes') {
+                $audioElem.data('isFirstPlay', 'yes');
+                $audioElem.one('ended', function () {
+                    $playBtn.removeClass('hide');
+                    $pauseBtn.addClass('hide');
+                    $audioElem.data('isFirstPlay', 'no');
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/menus/menu_view.js]: nice touch

**Comment by benrudolph** [comment]
bomb 💣 
...starting to think the web version is going to look better than the mobile version 😈

**Comment by dimagimon** [comment]
There are 9 errors:

* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 14 - 'define' is not defined. (no-undef)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 15 - 'define' is not defined. (no-undef)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 17 - 'exports' is not defined. (no-undef)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 51 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 59 - Expected indentation of 32 space characters but found 31. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 74 - Expected indentation of 28 space characters but found 29. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 76 - Expected indentation of 28 space characters but found 29. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 81 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 86 - Expected '===' and instead saw '=='. (eqeqeq)

**Comment by benrudolph** [comment]
test failure real: 
```
======================================================================
ERROR: test_compress_offline (corehq.apps.style.tests.test_compress_command.TestDjangoCompressOffline)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/style/tests/test_compress_command.py", line 69, in test_compress_offline
    call_command('compress', force=True)
  File "/vendor/lib/python2.7/site-packages/django/core/management/__init__.py", line 115, in call_command
    return klass.execute(*args, **defaults)
  File "/vendor/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/vendor/lib/python2.7/site-packages/django/core/management/base.py", line 533, in handle
    return self.handle_noargs(**options)
  File "/vendor/lib/python2.7/site-packages/compressor/management/commands/compress.py", line 302, in handle_noargs
    self.compress(sys.stdout, **options)
  File "/vendor/lib/python2.7/site-packages/compressor/management/commands/compress.py", line 259, in compress
    "%s" % (template.template_name, e))
CommandError: An error occurred during rendering /mnt/commcare-hq-ro/corehq/apps/preview_app/templates/preview_app/base.html: 'dragscroll/dragscroll.js' could not be found in the COMPRESS_ROOT '/mnt/commcare-hq-ro/staticfiles' or with staticfiles.
```

**Comment by dimagimon** [comment]
There are 9 errors:

* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 14 - 'define' is not defined. (no-undef)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 15 - 'define' is not defined. (no-undef)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 17 - 'exports' is not defined. (no-undef)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 51 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 59 - Expected indentation of 32 space characters but found 31. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 74 - Expected indentation of 28 space characters but found 29. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 76 - Expected indentation of 28 space characters but found 29. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 81 - Expected indentation of 12 space characters but found 13. (indent)
* corehq/apps/preview_app/static/preview_app/js/dragscroll.js, line 86 - Expected '===' and instead saw '=='. (eqeqeq)

File: corehq/apps/preview_app/static/preview_app/js/dragscroll.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                    cont.mm = function(e) {
+                        if (pushed) {
+                            (scroller = el.scroller||el).scrollLeft -=
+                                (- lastClientX + (lastClientX=e.clientX));
+                            scroller.scrollTop -=
+                                (- lastClientY + (lastClientY=e.clientY));
+                        }
+                    }, 0
+                );
+             })(dragged[i++]);
```
Comment (dimagimon) [inline corehq/apps/preview_app/static/preview_app/js/dragscroll.js]: Expected indentation of 12 space characters but found 13. (indent)

**Pull request merged**